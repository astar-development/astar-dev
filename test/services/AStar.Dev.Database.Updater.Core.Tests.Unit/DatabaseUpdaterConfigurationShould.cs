using AStar.Dev.Utilities;

namespace AStar.Dev.Database.Updater.Core;

[TestSubject(typeof(DatabaseUpdaterConfiguration))]
public class DatabaseUpdaterConfigurationShould
{
    [Fact]
    public void ContainTheExpectedProperties() =>
        new DatabaseUpdaterConfiguration
        {
            RootDirectory           = "Mock Root Directory",
            HonourFirstDelay        = true,
            MappingsFilePath        = "Mock Mappings File Path",
            SoftDeleteScheduledTime = TimeOnly.Parse("01:00:00"),
            HardDeleteScheduledTime = TimeOnly.Parse("02:00:00"),
            NewFilesScheduledTime   = TimeOnly.Parse("03:00:00")
        }.ToJson().ShouldMatchApproved();
}
