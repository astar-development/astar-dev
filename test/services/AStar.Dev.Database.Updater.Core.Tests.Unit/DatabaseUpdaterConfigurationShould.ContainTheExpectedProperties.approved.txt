{
  "rootDirectory": "Mock Root Directory",
  "honourFirstDelay": true,
  "mappingsFilePath": "Mock Mappings File Path",
  "softDeleteScheduledTime": "01:00:00",
  "hardDeleteScheduledTime": "02:00:00",
  "newFilesScheduledTime": "03:00:00"
}